x-pack/solutions/search/plugins/search_connectors/server/task.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger, CoreStart, SavedObjectsClient } from '@kbn/core/server';
+
+import type {
+  ConcreteTaskInstance,
+  TaskManagerStartContract,
+  TaskInstance,
+} from '@kbn/task-manager-plugin/server';
+
+import { LicensingPluginStart } from '@kbn/licensing-plugin/server';
+import type {
+  SearchConnectorsPluginStartDependencies,
+  SearchConnectorsPluginSetupDependencies,
+} from './types';
+import {
+  AgentlessConnectorsInfraService,
+  getConnectorsWithoutPolicies,
+  getPoliciesWithoutConnectors,
+} from './services';
+
+import { SearchConnectorsConfig } from './config';
+
+const AGENTLESS_CONNECTOR_DEPLOYMENTS_SYNC_TASK_ID = 'search:agentless-connectors-sync-task';
+const AGENTLESS_CONNECTOR_DEPLOYMENTS_SYNC_TASK_TYPE = 'search:agentless-connectors-sync';
```

Comment:
"Sync" is already taken in the connector namespace and it related to fetching data from a 3rd party, I think we can make it clearer by using a "manager" for the background task?

```suggestion
const AGENTLESS_CONNECTOR_DEPLOYMENTS_SYNC_TASK_TYPE = 'search:agentless-connectors-manager';
```